### PR TITLE
Fix float/integer handling in image API

### PR DESCRIPTION
**Pull request by MorrisJobke**
* IImage::crop/preciseResize now have type hinting for integers
* found while testing strict typing for PHP 7+(#7392)

### CODE REVIEW

File: lib/private/legacy/image.php [hunk 1]
```diff
@@ -550,6 +550,13 @@ public function loadFromFileHandle($handle) {
 	 * @return bool|resource An image resource or false on error
 	 */
 	public function loadFromFile($imagePath = false) {
+		try {
+			// detect if it is a path or maybe the images as string
+			// needed because the constructor iterates over all load* methods
+			realpath($imagePath);
```
Comment (MorrisJobke) [inline lib/private/legacy/image.php]: Have a look into the `load()` method of this class to realise how weird this works. You can pipe a resource, string with path, string with file content and string with base64 encoded file content into the constructor and then it iterates over all the methods to load it. 💥 This is not how an object should be initialised 🙈

Comment (rullzer) [reply lib/private/legacy/image.php]: Ah yes... it hurts every time I see that file... but I'm afraid to fix it :P

**Review by rullzer — APPROVED** [review]
Fine by me

**Comment by MorrisJobke** [comment]
I just fixed the failing tests.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**